src/lib/session-storage.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * This yields either sessionStorage, or when that particular
+ * API is not available, yields an object with limited API
+ * that behaves _like_ sessionStorage for the duration of
+ * the app, but does not persist across hard reloads.
+ */
+const storage = (typeof sessionStorage !== "undefined") ? sessionStorage : {
```

Comment:
pretty sure you can't actually do `typeof sessionStorage` because it throws immediately as soon as you try to access `window.sessionStorage`. Instead, you might want to try:
```js
let storage;
try {
  storage = sessionStorage;
  if(!sessionStorage) { // I think in some browsers it might not throw
     throw new Error("Session storage not available");
  }
} catch(err) {
  console.error("Session storage not accessible, using shim instead. Error: ", err);
  storage = {
    map: {},
    getItem: (name => storage.map[name]),
    setItem: ((name,value) => storage.map[name]=value)
  };
}
```